File: lib/src/main/java/com/easy/lib/weight/banner/BannerAdapter.java
package com.easy.lib.weight.banner;

import android.view.View;
import android.view.ViewGroup;

import androidx.viewpager.widget.PagerAdapter;

import java.util.ArrayList;
import java.util.List;

class BannerAdapter extends PagerAdapter {

    private List bannerList;
    private LoopBannerHolder loopBannerHolder;
    private int DAYTIME = 500;

    public BannerAdapter() {
        bannerList = new ArrayList();
    }

    public void setLoop(boolean loop) {
        if (loop) {
            DAYTIME = DAYTIME * 2;
        }
    }

    public void setData(List data) {
        bannerList.clear();
        bannerList.addAll(data);
    }

    public void createHolder(LoopBannerHolder holder) {
        loopBannerHolder = holder;
    }


    @Override
    public int getCount() {
        return bannerList.size() * DAYTIME;
    }

    @Override
    public boolean isViewFromObject(View view, Object object) {
        return view == object;
    }

    @Override
    public Object instantiateItem(ViewGroup container, int position) {
        //得到真实的position
        int realPosition = position % bannerList.size();
        //得到复用的View
        View view = loopBannerHolder.bindView(realPosition, bannerList.get(realPosition));
        container.addView(view);
        return view;
    }

    @Override
    public void destroyItem(ViewGroup container, int position, Object object) {
        View view = (View) object;
        container.removeView(view);
        BannerHelper.getInstance().getBannerViewMap().put(view, false);
    }

    public int getRealCount() {
        return bannerList.size();
    }


}

File: lib/src/main/java/com/easy/lib/weight/banner/BannerAdapter.kt
package com.easy.lib.weight.banner

import android.view.View
import android.view.ViewGroup
import androidx.viewpager.widget.PagerAdapter
import kotlin.collections.ArrayList

internal class BannerAdapter : PagerAdapter() {
    private val bannerList = ArrayList<Any>()

    private var loopBannerHolder: LoopBannerHolder? = null
    private var DAYTIME = 500
    fun setLoop(loop: Boolean) {
        if (loop) {
            DAYTIME = DAYTIME * 2
        }
    }

    fun <T: Any> setData(data: ArrayList<T>) {
        bannerList.clear()
        bannerList.addAll(data)
    }

    fun createHolder(holder: LoopBannerHolder) {
        loopBannerHolder = holder
    }

    override fun getCount(): Int {
        return bannerList.size * DAYTIME
    }

    override fun isViewFromObject(
        view: View,
        `object`: Any
    ): Boolean {
        return view === `object`
    }

    override fun instantiateItem(
        container: ViewGroup,
        position: Int
    ): Any { //得到真实的position
        val realPosition = position % bannerList.size
        //得到复用的View
        val view =
            loopBannerHolder!!.bindView(realPosition, bannerList[realPosition] as ArrayList<*>)
        container.addView(view)
        return view
    }

    override fun destroyItem(
        container: ViewGroup,
        position: Int,
        `object`: Any
    ) {
        val view = `object` as View
        container.removeView(view)
        BannerHelper.bannerViewMap.set(view,false)

    }

    val realCount: Int
        get() = bannerList.size

}